model/proto/keyvalue.proto

```diff
@@ -0,0 +1,27 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+syntax = "proto3";
+
+package elastic.apm.v1;
+
+option go_package = "github.com/elastic/apm-data/model/modelpb";
+
+message KeyValue {
+  string key = 1;
+  string value = 2;
```

Comment:
**issue**: I don't think this works. `Struct` is a lot more than strings.

This is restricting the value a "value" can assume.

Before it was `any` but now it's `string`.

Reply:
That makes sense. I've switched the string to a `google.protobuf.Value`.

```
Running benchstat
goos: darwin
goarch: arm64
pkg: github.com/elastic/apm-data/codec
                         │ bench-main  │             bench-branch             │
                         │   sec/op    │    sec/op     vs base                │
Encode/format=json-10      22.19µ ± 2%   21.63µ ±  5%   -2.54% (p=0.011 n=10)
Encode/format=vtproto-10   2.509µ ± 3%   1.362µ ±  2%  -45.72% (p=0.000 n=10)
Decode/format=json-10      17.93µ ± 4%   16.23µ ± 10%   -9.49% (p=0.007 n=10)
Decode/format=vtproto-10   7.014µ ± 4%   6.465µ ±  5%   -7.83% (p=0.001 n=10)
geomean                    9.148µ        7.456µ        -18.49%

                         │  bench-main  │            bench-branch             │
                         │   bytes/op   │   bytes/op    vs base               │
Encode/format=json-10      6.169Ki ± 0%   5.988Ki ± 0%  -2.93% (p=0.000 n=10)
Encode/format=vtproto-10   2.710Ki ± 0%   2.718Ki ± 0%  +0.29% (p=0.000 n=10)
geomean                    4.089Ki        4.034Ki       -1.33%

pkg: github.com/elastic/apm-data/input/elasticapm/internal/decoder
                   │  bench-main  │         bench-branch          │
                   │    sec/op    │   sec/op     vs base          │
NDStreamDecoder-10   44.32n ± 14%   44.26n ± 6%  ~ (p=0.739 n=10)

pkg: github.com/elastic/apm-data/model/modelpb
                          │ bench-main  │         bench-branch          │
                          │   sec/op    │   sec/op     vs base          │
APMEventToJSON/to-json-10   11.58µ ± 1%   11.52µ ± 2%  ~ (p=0.579 n=10)
```